parcels/field.py

```diff
@@ -428,7 +428,9 @@ def gradient(self, update=False, tindex=None):
     def search_indices_vertical_z(self, z):
         grid = self.grid
         z = np.float32(z)
-        if z < grid.depth[0] or z > grid.depth[-1]:
+        if z < grid.depth[0]:
+            raise FieldOutOfBoundError_Surface(0, 0, z, field=self)
```

Comment:
`FieldOutOfBounrError_Surface` should be also in `search_indices_vertical_s` (line 476)